var/spack/repos/builtin/packages/flexi/package.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Flexi(CMakePackage):
+    """Open Source High-Order Unstructured Discontinuous Galerkin Fluid
+    Dynamics Solver"""
+
+    homepage = "https://www.flexi-project.org/"
+    git      = "https://github.com/flexi-framework/flexi.git"
+
+    version('master')
+
+    variant('mpi', default=True, description='Enable MPI')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('hdf5+fortran+mpi', when='+mpi')
+    depends_on('hdf5+fortran~mpi', when='~mpi')
+    depends_on('lapack')
+    depends_on('zlib')
+
+    patch('https://github.com/flexi-framework/flexi/pull/6.patch', sha256='1c8fc140a9c83dbd77eb3e1d71ed18a35c3b67d0e9fbf75da03acab25c60b2c5')
+
+    def flag_handler(self, name, flags):
+        if name == 'fflags':
+            if self.spec.satisfies('%gcc@10:'):
+                if flags is None:
+                    flags = []
+                flags.append('-fallow-argument-mismatch')
+
+        return (flags, None, None)
+
+    def cmake_args(self):
+        args = [
+            '-DFLEXI_BUILD_HDF5=OFF'
+        ]
```

Comment:
We can:
```suggestion
        args = [
            '-DFLEXI_BUILD_HDF5=OFF',
            self.define_from_variant('FLEX_MPI', 'mpi')
        ]
```
and delete the lines below.